src/Sentry/Protocol/ProfileInfo.cs

```diff
@@ -0,0 +1,86 @@
+using Sentry.Extensibility;
+using Sentry.Internal;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol;
+
+/// <summary>
+/// Profiling context information.
+/// </summary>
+internal sealed class ProfileInfo : IJsonSerializable
+{
+    /// <summary>
+    /// Profile's event ID.
+    /// </summary>
+    public SentryId EventId { get; private set; } = SentryId.Create();
+
+    public DebugMeta DebugMeta { get; set; } = new() { Images = new() };
+
+    private readonly Contexts _contexts = new();
+
+    /// <inheritdoc />
+    public Contexts Contexts
+    {
+        get => _contexts;
+        set => _contexts.ReplaceWith(value);
+    }
+
+    public SampleProfile Profile { get; set; } = new();
+
+    public DateTimeOffset StartTimestamp { get; set; } = DateTimeOffset.UtcNow;
+
+    public string? Environment { get; set; }
+
+    public string? Platform { get; set; } = Constants.Platform;
+
+    public string? Release { get; set; }
+
+    public Transaction? Transaction { get; set; }
+
+
+    public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
+    {
+        // NOTE: some values are required by the protocol so we pass "" if we don't have them.
+
+        writer.WriteStartObject();
+
+        writer.WriteString("version", "1");
+        writer.WriteSerializable("event_id", EventId, logger);
+        writer.WriteString("timestamp", StartTimestamp);
+        writer.WriteStringIfNotWhiteSpace("platform", Platform);
+        writer.WriteStringIfNotWhiteSpace("release", Release);
+        writer.WriteStringIfNotWhiteSpace("environment", Environment);
+        writer.WriteSerializable("debug_meta", DebugMeta, logger);
+
+        // TODO writer.WriteSerializable("device", _contexts.Device, logger);
+        //  https://github.com/getsentry/relay/blob/master/relay-profiling/src/sample.rs#L117
+        writer.WriteStartObject("device");
+        writer.WriteString("architecture", _contexts.Device.Architecture ?? "");
+        writer.WriteStringIfNotWhiteSpace("manufacturer", _contexts.Device.Manufacturer);
+        writer.WriteStringIfNotWhiteSpace("model", _contexts.Device.Model);
+        writer.WriteEndObject();
+
+        // TODO writer.WriteSerializable("os", _contexts.OperatingSystem, logger);
+        //  https://github.com/getsentry/relay/blob/master/relay-profiling/src/sample.rs#L102
+        var rawOS = _contexts.OperatingSystem.RawDescription?.Replace("Microsoft Windows", "Windows")?.Split(' ', 2);
```

Comment:
We parse this in Relay: https://github.com/getsentry/relay/blob/1a7f256f65a51da069959adf3eb5619c63660af4/relay-general/src/store/normalize/contexts.rs#L330